src/SystemInterface.py

```diff
@@ -106,3 +106,56 @@ def monitor_journalctl_for_message(self, service_name: str, message: str,
                 process.terminate()
             except:
                 pass
+
+    def restart_component(self, component_name: str, reset: bool):
+        if reset:
+            try:
+                cmd = [
+                    "sudo",
+                    "systemctl",
+                    "reset-failed"
+                ]
+
+                # Run the command and stream output
+                process = subprocess.Popen(
+                    cmd,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    universal_newlines=True,
+                )
+            except Exception as e:
+                print(f"Error: {e}")
+            finally:
+                # Ensure process is terminated
+                try:
+                    process.terminate()
+                except:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```

Similar issue at line number 160.